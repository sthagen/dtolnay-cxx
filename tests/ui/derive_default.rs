#[cxx::bridge]
mod ffi {
    #[derive(Default)]
    enum NoDefault {
        Two,
        Three,
        Five,
        Seven,
    }

    #[derive(Default)]
    enum MultipleDefault {
        #[default]
        Two,
        Three,
        Five,
        #[default]
        Seven,
    }
}

#[cxx::bridge]
mod ffi2 {
    #[derive(Default)]
    enum BadDefault {
        #[default(repr)]
        Two,
        #[default = 3]
        Three,
    }
}

fn main() {}
