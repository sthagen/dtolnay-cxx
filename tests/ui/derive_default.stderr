error: derive(Default) on enum requires exactly one variant to be marked with #[default]
 --> tests/ui/derive_default.rs:3:14
  |
3 |     #[derive(Default)]
  |              ^^^^^^^

error: derive(Default) on enum requires exactly one variant to be marked with #[default] (found 2)
  --> tests/ui/derive_default.rs:11:14
   |
11 |     #[derive(Default)]
   |              ^^^^^^^

error: #[default] attribute does not accept an argument
  --> tests/ui/derive_default.rs:26:18
   |
26 |         #[default(repr)]
   |                  ^

error: #[default] attribute does not accept an argument
  --> tests/ui/derive_default.rs:28:19
   |
28 |         #[default = 3]
   |                   ^
